packages/compass-e2e-tests/helpers/assistant-service.ts

```diff
@@ -0,0 +1,248 @@
+        id: responseId,
+        object: 'realtime.response',
+        status: 'in_progress',
+        output: [],
+        usage: {
+          input_tokens: 0,
+          output_tokens: 0,
+          total_tokens: 0,
+        },
+      },
+      sequence_number: sequenceNumber++,
+    })}\n\n`
+  );
+
+  // Send output_item.added event
+  res.write(
+    `data: ${JSON.stringify({
+      type: 'response.output_item.added',
+      response_id: responseId,
+      output_index: 0,
+      item: {
+        id: itemId,
+        object: 'realtime.item',
+        type: 'message',
+        role: 'assistant',
+        content: [],
+      },
+      sequence_number: sequenceNumber++,
+    })}\n\n`
+  );
+
+  // Send the content in chunks
+  const words = content.split(' ');
+  let index = 0;
+
+  const sendChunk = () => {
+    if (index < words.length) {
+      const word = words[index] + (index < words.length - 1 ? ' ' : '');
+      // Send output_text.delta event
+      res.write(
+        `data: ${JSON.stringify({
+          type: 'response.output_text.delta',
+          response_id: responseId,
+          item_id: itemId,
+          output_index: 0,
+          delta: word,
+          sequence_number: sequenceNumber++,
+        })}\n\n`
+      );
+      index++;
+      setTimeout(sendChunk, 10);
+    } else {
+      // Send output_item.done event
+      res.write(
+        `data: ${JSON.stringify({
+          type: 'response.output_item.done',
+          response_id: responseId,
+          output_index: 0,
+          item: {
+            id: itemId,
+            object: 'realtime.item',
+            type: 'message',
+            role: 'assistant',
+            content: [
+              {
+                type: 'text',
+                text: content,
+              },
+            ],
+          },
+          sequence_number: sequenceNumber++,
+        })}\n\n`
+      );
+
+      // Send response.completed event
+      const tokenCount = Math.ceil(content.split(' ').length * 1.3);
+      res.write(
+        `data: ${JSON.stringify({
+          type: 'response.completed',
+          response: {
+            id: responseId,
+            object: 'realtime.response',
+            status: 'completed',
+            output: [
+              {
+                id: itemId,
+                object: 'realtime.item',
+                type: 'message',
+                role: 'assistant',
+                content: [
+                  {
+                    type: 'text',
+                    text: content,
+                  },
+                ],
+              },
+            ],
+            usage: {
+              input_tokens: 10,
+              output_tokens: tokenCount,
+              total_tokens: 10 + tokenCount,
+            },
+          },
+          sequence_number: sequenceNumber++,
+        })}\n\n`
+      );
+
+      res.write('data: [DONE]\n\n');
+      res.end();
+    }
+  };
+
+  sendChunk();
+}
+
+export async function startMockAssistantServer(
```

Comment:
So.. Usually E2E tests are purposefully integration tests, ie. we try not to mock or stub anything if at all possible. But I do get that the assistant is an expensive feature to be running many times on every PR and it is also particularly non-deterministic.

So I think in this case it is OK, just something to keep in mind going forward.